apps/seqan_flexbar/adapter_trimming.h

```diff
@@ -87,12 +87,17 @@ struct Mode {};
 struct Auto : Mode {};
 
 //Tagging struct representing the the match algorithm working
-//with values supplied by the user. Saves those values as members.
+//with values supplied by the user. Saves those  values as members.
```

Comment:
OMG This was extremely ugly :scream:

Is it much work to change it? I have no clue as it's the first time that I look into this app... :weary: 
I wouldn't do any cast nor pointer/new. Maybe use only the struct User for options and add a boolean auto to it? Where is tag dispatching needed?